scripts/setupTypeScriptRollup.js

```diff
@@ -0,0 +1,264 @@
+	}
+].forEach(({ view, vars, contextModule }) => {
+	const svelteFilePath = path.join(projectRoot, 'src', `${view}.svelte`);
+	let file = fs.readFileSync(svelteFilePath, 'utf8');
+
+	file = file.replace(/(?:<script)(( .*?)*?)>/gm, '<script$1 lang="ts">');
+
+	if (vars) {
+		vars.forEach(({ name, type }) => {
+			file = file.replace(
+				`export let ${name};`,
+				`export let ${name}: ${type};`
+			);
+		});
+	}
+
+	if (contextModule) {
+		contextModule.forEach(({ js, ts }) => {
+			file = file.replace(`${js}`, `${ts}`);
+		});
+	}
+
+	fs.writeFileSync(svelteFilePath, file);
+});
+
+// Edit rollup config
+const rollupConfigPath = path.join(projectRoot, 'rollup.config.js');
+let rollupConfig = fs.readFileSync(rollupConfigPath, 'utf8');
+
+// Edit imports
+rollupConfig = rollupConfig.replace(
+	`'rollup-plugin-terser';`,
+	`'rollup-plugin-terser';
+import sveltePreprocess from 'svelte-preprocess';
+import typescript from '@rollup/plugin-typescript';`
+);
+
+// Edit inputs
+rollupConfig = rollupConfig.replace(
+	`onwarn(warning);`,
+	`(warning.code === 'THIS_IS_UNDEFINED') ||\n\tonwarn(warning);`
+);
+rollupConfig = rollupConfig.replace(
+	`input: config.client.input(),`,
+	`input: config.client.input().replace(/\.js$/, '.ts'),`
+);
+rollupConfig = rollupConfig.replace(
+	`input: config.server.input(),`,
+	`input: { server: config.server.input().server.replace(/\.js$/, ".ts") },`
+);
+rollupConfig = rollupConfig.replace(
+	`input: config.serviceworker.input(),`,
+	`input: config.serviceworker.input().replace(/\.js$/, '.ts'),`
+);
+
+// Add preprocess to the svelte config, this is tricky because there's no easy signifier.
+// Instead we look for 'hydratable: true,'
+rollupConfig = rollupConfig.replace(
+	new RegExp('hydratable: true,', 'g'),
+	'hydratable: true,\n\t\t\t\tpreprocess: sveltePreprocess(),'
+);
+
+// Edit service worker
+const serviceWorkerConfigPath = path.join(
+	projectRoot,
+	'src',
+	'service-worker.ts'
+);
+
+let serviceWorkerConfig = fs.readFileSync(serviceWorkerConfigPath, 'utf8');
+
+serviceWorkerConfig = serviceWorkerConfig.replace(
+	`'@sapper/service-worker';`,
+	`'@sapper/service-worker';\n
+type Clientable = { clients: Clients };`
+);
+
+serviceWorkerConfig = serviceWorkerConfig.replace(
+	`shell.concat(files);`,
+	`(shell as string[]).concat(files as string[]);`
+);
+
+serviceWorkerConfig = serviceWorkerConfig.replace(
+	`const cached = new Set(to_cache);`,
+	`const cached = new Set(to_cache);\n
+const claimClient = (input: Clientable) => {
+	input.clients.claim();
+};
+
+const skipWaiting = (input: ServiceWorkerGlobalScope) => {
+	input.skipWaiting();
+};`
+);
+
+serviceWorkerConfig = serviceWorkerConfig.replace(
+	`'install', event =>`,
+	`'install', <EventType extends ExtendableEvent>(event: EventType) =>`
+);
+
+serviceWorkerConfig = serviceWorkerConfig.replace(
+	`self.skipWaiting();`,
+	`skipWaiting((self as any) as ServiceWorkerGlobalScope);`
```

Comment:
I'd probably prefer to do `((self as any) as ServiceWorkerGlobalScope).skipWaiting()` vs defining a new method